pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -97,5 +119,253 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
+		if !errors.IsNotFound(err) {
+			// Error reading the MasterUserRecord, requeue the request
+			return reconcile.Result{}, err
+		}
+	} else {
+		// If we successfully found an existing MasterUserRecord then our work here is done, set the status
+		// to Complete and return
+		statusError := r.setStatusComplete(instance, "")
+		if statusError != nil {
+			reqLogger.Error(statusError, "Error updating UserSignup Status to Complete", "Name", instance.Name)
+			return reconcile.Result{}, statusError
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// Check the user approval policy.
+	userApprovalPolicy, err := r.ReadUserApprovalPolicyConfig()
+	if err != nil {
+		statusError := r.setStatusFailedToReadUserApprovalPolicy(instance, err.Error())
+		if statusError != nil {
+			reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+			return reconcile.Result{}, statusError
+		}
+
+		reqLogger.Error(err, "Error reading user approval policy")
+		return reconcile.Result{}, err
+	}
+
+	// If the signup has been explicitly approved (by an admin), or the user approval policy is set to automatic,
+	// then proceed with the signup
+	if instance.Spec.Approved || userApprovalPolicy == config.UserApprovalPolicyAutomatic {
+		if instance.Spec.Approved {
+			statusError := r.setStatusApprovedByAdmin(instance, "")
+			if statusError != nil {
+				reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+				return reconcile.Result{}, statusError
+			}
+		}
+
+		if userApprovalPolicy == config.UserApprovalPolicyAutomatic {
+			statusError := r.setStatusApprovedAutomatically(instance, "")
+			if statusError != nil {
+				reqLogger.Error(statusError, "Error updating UserSignup Status", "Name", instance.Name)
+				return reconcile.Result{}, statusError
+			}
+		}
+
+		var targetCluster string
+
+		// If a target cluster hasn't been selected, select one from the members
+		if instance.Spec.TargetCluster != "" {
+			targetCluster = instance.Spec.TargetCluster
+		} else {
+			// Automatic cluster selection
+			members := cluster.GetMemberClusters()
+			if len(members) > 0 {
+				targetCluster = members[0].Name
+			} else {
+				statusError := r.setStatusNoClustersAvailable(instance, "No member clusters found")
+				if statusError != nil {
+					reqLogger.Error(statusError, "Error updating UserSignup Status", "UserID", instance.Spec.UserID)
+					return reconcile.Result{}, statusError
+				}
+
+				err = NewSignupError("No target clusters available")
+				reqLogger.Error(err, "No member clusters found")
+				return reconcile.Result{}, err
+			}
+		}
+
+		// Provision the MasterUserRecord
+		err = r.provisionMasterUserRecord(instance, targetCluster, reqLogger)
+		if err != nil {
+			// If the error is because the MUR already exists, delay the next reconcile loop
+			if errors.IsAlreadyExists(err) {
+				return reconcile.Result{
+					RequeueAfter: 1 * time.Hour,
```

Comment:
One hour? :) Why so long? IMO we don't need any delay at all. The next reconcile will update the status to complete and exit w/o error. Or just set the status to Complete here.

**Update**: You actually already set the status to complete in line 259, so this code won't be executed at all.

Reply:
Oops, I missed this in the refactor.  Fixed now.